web/src/features/modals/AnnouncementModal.tsx

```diff
@@ -0,0 +1,83 @@
+import { SourceColorAnnouncementImage } from 'icons/announcementSourceColorImage';
+import { useAtom } from 'jotai';
+import { atomWithStorage } from 'jotai/utils';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { HiXMark } from 'react-icons/hi2';
+import { hasOnboardingBeenSeenAtom } from 'utils/state/atoms';
+
+const hasAnnouncementBeenSeenAtom = atomWithStorage(
+  'announcementSeen',
+  localStorage.getItem('announcementSeen') ?? false
+);
+
+const handleTierGaterClick = (event: React.MouseEvent<HTMLDivElement>) => {
+  event.stopPropagation();
+  event.preventDefault();
+};
+
+export default function AnnouncementModal() {
+  const { t } = useTranslation();
+  const [hasAnnouncementBeenSeen, setHasAnnouncementBeenSeen] = useAtom(
+    hasAnnouncementBeenSeenAtom
+  );
+  const [hasOnboardingBeenSeen] = useAtom(hasOnboardingBeenSeenAtom);
+  const [hasOnboardingBeenSeenInSession, setHasOnboardingBeenSeenInSession] =
+    useState(false);
+  const isExpired = new Date() > new Date('2024-06-30');
+
+  // Check if onboarding has been seen in the current session
+  useEffect(() => {
+    const onboardingSeenInSession = sessionStorage.getItem('onboardingSeenInSession');
+    if (onboardingSeenInSession) {
+      setHasOnboardingBeenSeenInSession(true);
+    } else if (!hasOnboardingBeenSeen) {
+      sessionStorage.setItem('onboardingSeenInSession', 'true');
+      setHasOnboardingBeenSeenInSession(true);
+    }
+  }, [hasOnboardingBeenSeen]);
+
+  const handleDismiss = () => {
+    setHasAnnouncementBeenSeen(true);
+  };
+
+  if (
+    hasAnnouncementBeenSeen ||
+    isExpired ||
+    !hasOnboardingBeenSeen ||
+    hasOnboardingBeenSeenInSession
+  ) {
+    return null;
+  }
+
+  return (
+    <div
+      role="presentation"
+      onClick={handleDismiss}
+      onKeyDown={handleDismiss}
+      className="absolute left-0 top-0 z-50 flex h-full w-full flex-col items-center justify-center bg-black/20"
```

Comment:
Why could we not reuse the Modal component, was the design too different?

Reply:
I also think it's a bit of a shame to not use [Radix modal](https://www.radix-ui.com/primitives/docs/components/dialog) here which is much more accessible :)

Reply:
On second thought it's probably not too critical for this short-lived one, so I'm also okay with sticking to your current implementation for now if this becomes blocking :)

Reply:
The design was too different to reuse it. I can try to use the Radix's, if you would like. What is the reason why it is smart to use?

Reply:
Okay, fair :) Here's why it's nice (in addition to being tested in a multitude of browsers and environments):

![Screenshot 2024-06-04 at 23 37 46](https://github.com/electricitymaps/electricitymaps-contrib/assets/3296643/6a773cfd-cdea-45d1-a550-4581af739c08)

With that said I'm also fine with not during it here, but next time it would be good to revamp a generic Modal component that we can more easily reuse :)